cmd/create/ocmrole/cmd.go

```diff
@@ -308,46 +355,48 @@ func createRoles(reporter *rprtr.Object, awsClient aws.Client, prefix string, po
 		return roleARN, nil
 	}
 	reporter.Debugf("Creating role '%s'", roleName)
+
+	iamTags := map[string]string{
+		tags.RolePrefix:  prefix,
+		tags.RoleType:    aws.OCMRole,
+		tags.Environment: env,
+	}
+
 	roleARN, err = awsClient.EnsureRole(roleName, string(policy), permissionsBoundary,
-		"", map[string]string{
-			tags.RolePrefix:  prefix,
-			tags.RoleType:    aws.OCMRole,
-			tags.Environment: env,
-		})
+		"", iamTags)
 	if err != nil {
 		return "", err
 	}
 	reporter.Infof("Created role '%s' with ARN '%s'", roleName, roleARN)
 
+	// create and attach the permission policy to the role
 	filename = fmt.Sprintf("sts_%s_permission_policy.json", aws.OCMRolePolicyFile)
-	path = fmt.Sprintf("templates/policies/%s", filename)
-
-	policy, err = aws.ReadPolicyDocument(path)
+	err = createPermissionPolicy(reporter, awsClient, policyARN, iamTags, roleName, filename)
 	if err != nil {
 		return "", err
 	}
 
-	reporter.Debugf("Creating permission policy '%s'", policyARN)
-	policyARN, err = awsClient.EnsurePolicy(policyARN, string(policy),
-		"", map[string]string{
-			tags.RolePrefix:  prefix,
-			tags.RoleType:    aws.OCMRolePolicyFile,
-			tags.Environment: env,
-		})
-	if err != nil {
-		return "", err
-	}
+	if isAdmin {
+		// tag role with admin tag
+		err = awsClient.AddRoleTag(roleName, tags.AdminRole, "true")
```

Comment:
Why not add the tag directly to the `iamTags` map and avoid an extra API call?

Reply:
I find the code cleaner like that when all admin changes are in one place, like keeping a separate policy, that we also tag with the admin tag (and we don't tag the regular permission policy with it).